Jenkinsfile

```diff
@@ -1,52 +1,135 @@
   }
-  environment {
-    REPO = 'lib_logging'
-    VIEW = getViewName(REPO)
-  }
-  options {
-    skipDefaultCheckout()
-  }
-  stages {
-    stage('Get view') {
-      steps {
-        xcorePrepareSandbox("${VIEW}", "${REPO}")
-      }
-    }
-    stage('Library checks') {
-      steps {
-        xcoreLibraryChecks("${REPO}")
-      }
-    }
-    stage('Tests') {
-      steps {
-        runXmostest("${REPO}", 'tests')
-      }
-    }
-    stage('xCORE builds') {
-      steps {
-        dir("${REPO}") {
-          xcoreAllAppsBuild('examples')
-          xcoreAllAppNotesBuild('examples')
-          dir("${REPO}") {
-            runXdoc('doc')
+  stages{
+    stage ('Full test'){
+      parallel {
+        stage('Standard Run') {
+          agent {
+            label 'x86_64&&brew'
+          }
+          environment {
+            REPO = 'lib_logging'
+            VIEW = getViewName(REPO)
+          }
+          options {
+            skipDefaultCheckout()
+          }
+          stages{
+            stage('Get view') {
+              steps {
+                sh 'rm -f cur_view.txt' //Temp hack. Need to work out why post/cleanup is not doing its job
+                xcorePrepareSandbox("${VIEW}", "${REPO}")
+              }
+            }
+            stage('Library checks') {
+              steps {
+                xcoreLibraryChecks("${REPO}")
+              }
+            }
+            stage('Tests') {
+              steps {
+                runXmostest("${REPO}", 'legacy_tests')
+              }
+            }
+            stage('xCORE builds') {
+              steps {
+                dir("${REPO}") {
+                  xcoreAllAppsBuild('examples')
+                  xcoreAllAppNotesBuild('examples')
+                  dir("${REPO}") {
+                    runXdoc('doc')
+                  }
+                }
+              }
+            }
+          }
+          post {
+            success {
+              updateViewfiles()
+            }
+            cleanup {
+              xcoreCleanSandbox()
+            }
           }
         }
-      }
-    }
-  }
-  post {
-    success {
-      updateViewfiles()
-    }
-    cleanup {
-      xcoreCleanSandbox()
-    }
-  }
+        stage('XCORE AI tests') {
+          agent {
+            label "srv-bri-nuc2"
+          }
+          environment {
+            // '/XMOS/tools' from get_tools.py and rest from tools installers
+            TOOLS_PATH = "/XMOS/tools/${params.AI_TOOLS_VERSION}/XMOS/xTIMEcomposer/${params.AI_TOOLS_VERSION}"
+            REPO = 'lib_logging'
+            VIEW = getViewName(REPO)
+          }
+          options {
+            skipDefaultCheckout()
+          }
+          stages {
+            stage('Checkout') {
+              steps {
+                checkout scm
+              }
+            }
+            stage('Install Dependencies') {
+              steps {
+                sh '/XMOS/get_tools.py ' + params.AI_TOOLS_VERSION
+                installDependencies()
```

Comment:
I would change this to installPipfile since the nucs don't have brew installed and we've removed the need for cpanfiles.